integration/kotlinx-coroutines-firebase-android/kotlinx-coroutines-firebase-android/src/main/java/kotlinx/coroutines/experimental/firebase/android/DatabaseReference.kt

```diff
@@ -0,0 +1,266 @@
+package kotlinx.coroutines.experimental.firebase.android
+
+import com.google.firebase.FirebaseException
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.GenericTypeIndicator
+import com.google.firebase.database.Query
+import com.google.firebase.database.ValueEventListener
+import kotlinx.coroutines.experimental.suspendCancellableCoroutine
```

Comment:
Having a sample project is good, but it's not something that can easily be integrated into CI and depends on external service.
We should have unit tests which check every integration aspect but are run regularly with `./gradlew tests`.

And then sample app may be rewritten as a simple UI application which shows some data from Firebase in a reactive way instead of unit tests relying on the availability of `coroutines-testing.firebaseio.com`

Reply:
Even better idea for sample project: https://github.com/firebase/quickstart-java/tree/master/database/src/main/java/com/google/firebase/quickstart

Firebase quickstart, which shows integration advantages.
But it's an extra mile, I can do it myself if you don't want to